precompiles/dapps-staking/DappsStaking.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+pragma solidity >=0.7.0;
+
+/// Interface to the precompiled contract on Shibuya/Shiden/Astar
+/// Predeployed at the address 0x0000000000000000000000000000000000005001
+interface DappsStaking {
+
+    // Storage getters
+
+    /// @dev Read current era.
+    /// @return The current era
+    function read_current_era() external view returns (uint256);
+
+    /// @dev Read Total network reward for the given era
+    /// @return Total network reward for the given era
+    function read_era_reward(uint32 era) external view returns (uint128);
+
+    /// @dev Read Total staked amount for the given era
+    /// @return Total staked amount for the given era
+    function read_era_staked(uint32 era) external view returns (uint128);
+
+    /// @dev Read Locked amount for the staker
+    /// @return Locked amount for the staker
+    function read_ledger(address staker) external view returns (uint128);
+
+    /// @dev Read stakers on the contract in the given era
+    /// @return The array of stakers in the given era
+    function read_contract_era_stakers(address contract_id, uint32 era) external view returns (uint256[] memory);
```

Comment:
If this is information we need to have, I will have to change the Individual claim implementation so that we can read it directly. Otherwise the call will be super expensive.